broken with debug@4

```
  when preloaded with -r, automatically logs all debug calls with log level debug to a default pino logger
not ok Cannot find module 'debug/src/debug.js'
    stack: |
      Test.tap.afterEach (test/index.js:25:15)
    at:
      line: 548
      column: 15
      file: module.js
      function: Function.Module._resolveFilename
    code: MODULE_NOT_FOUND
    test: >-
      when preloaded with -r, automatically logs all debug calls with log level
      debug to a default pino logger
```

from pino-integration

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,8 +2,14 @@ node_js:
 - "4"
 - "6"
 - "7"
+env:
+- DEBUG_VERSION=2
+- DEBUG_VERSION=3
+- DEBUG_VERSION=4
 sudo: false
 language: node_js
-script: "npm run ci"
+script:
+  - npm install --no-save debug@^$DEBUG_VERSION
+  - npm run ci
 after_success:
   - npm i -g codecov && codecov -t a62849b5-9006-4dd4-8334-94624d6d8ea2

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,8 @@
 node_js:
 - "4"
 - "6"
-- "7"
+- "8"
+- "10"
 env:
 - DEBUG_VERSION=2
 - DEBUG_VERSION=3

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,6 +7,10 @@ env:
 - DEBUG_VERSION=2
 - DEBUG_VERSION=3
 - DEBUG_VERSION=4
+matrix:
+  exclude:
+  - node_js: "4"
+    env: DEBUG_VERSION=4
 sudo: false
 language: node_js
 script:

--- a/index.js
+++ b/index.js
@@ -67,7 +67,7 @@ function override (script) {
         return Object.setPrototypeOf(function pinoDebugWrappedRequire(s) {
           var dirname = __dirname.split(pinoDebugPath.sep)
           var lastNodeModulesIndex = dirname.lastIndexOf('node_modules')
-          var isDebug = lastNodeModulesIndex > 0 && dirname[lastNodeModulesIndex + 1] === 'debug'
+          var isDebug = lastNodeModulesIndex >= 0 && dirname[lastNodeModulesIndex + 1] === 'debug'
           var pathToPinoDebug = '${pathToPinoDebug}'
 
           if (isDebug) {

--- a/index.js
+++ b/index.js
@@ -62,24 +62,37 @@ function override (script) {
   var head = `(function(exports, require, module, __filename, __dirname) {
       require = (function (req) {
         var pinoDebugOs = require('os')
+        var pinoDebugPath = require('path')
         var Object = ({}).constructor
         return Object.setPrototypeOf(function pinoDebugWrappedRequire(s) {
-          var dirname = __dirname.slice(-22)
+          var dirname = __dirname.split(pinoDebugPath.sep)
+          var lastNodeModulesIndex = dirname.lastIndexOf('node_modules')
+          var isDebug = lastNodeModulesIndex > 0 && dirname[lastNodeModulesIndex + 1] === 'debug'
           var pathToPinoDebug = '${pathToPinoDebug}'
 
-          if (pinoDebugOs.platform() === 'win32') {
-              dirname = dirname.replace(/\\\\/g, '/')
-          }
-
-          if (s === './debug' && /node_modules\\/debug/.test(dirname)) {
+          if (isDebug) {
             var dbg = req(pathToPinoDebug)
             var real = req(s)
-            Object.assign(dbg, real)
-            Object.defineProperty(real, 'save', {get: function () {
-              Object.defineProperty(real, 'save', {value: dbg.save})
-              return real.save
-            }, configurable: true})
-            return dbg
+            if (s === './common') {
+              var wrapped = function pinoDebugWrapSetup(env) {
+                var orig = real(env)
+                Object.assign(dbg, orig)
+                Object.defineProperty(orig, 'save', {get: function () {
+                  Object.defineProperty(orig, 'save', {value: dbg.save})
+                  return orig.save
+                }, configurable: true})
+                return dbg
+              }
+              return wrapped
+            }
+            if (s === './debug') {
+              Object.assign(dbg, real)
+              Object.defineProperty(real, 'save', {get: function () {
+                Object.defineProperty(real, 'save', {value: dbg.save})
+                return real.save
+              }, configurable: true})
+              return dbg
+            }
           }
           return req(s)
         }, req)

--- a/package.json
+++ b/package.json
@@ -31,8 +31,10 @@
   ],
   "license": "MIT",
   "dependencies": {
-    "debug": "^3.1.0",
     "pino": "^4.0.2"
+  },
+  "peerDependencies": {
+    "debug": ">=2"
   },
   "devDependencies": {
     "cross-env": "^5.2.0",

--- a/test/index.js
+++ b/test/index.js
@@ -4,31 +4,47 @@ var tap = require('tap')
 var through = require('through2')
 var test = tap.test
 
+const debugModules = [
+  // <= 2.4
+  [
+    'debug/node.js',
+    'debug/debug.js'
+  ],
+  // <= 4.0.1
+  [
+    'debug/node.js',
+    'debug/src/node.js',
+    'debug/src/debug.js'
+  ],
+  [
+    'debug/src/node.js',
+    'debug/src/common.js'
+  ]
+]
+
+const commonModules = [
+  'debug',
+  '../debug',
+  '../',
+  './'
+]
+
 tap.afterEach((done) => {
-  var deps
-  try {
-    // debug 2.4 and down
-    deps = [
-      require.resolve('debug'),
-      require.resolve('debug/node.js'),
-      require.resolve('debug/debug.js'),
-      require.resolve('../debug'),
-      require.resolve('../'),
-      require.resolve('./')
-    ]
-  } catch (e) {
-    // debug 2.5 and up
-    deps = [
-      require.resolve('debug'),
-      require.resolve('debug/node.js'),
-      require.resolve('debug/src/node.js'),
-      require.resolve('debug/src/debug.js'),
-      require.resolve('../debug'),
-      require.resolve('../'),
-      require.resolve('./')
-    ]
+  let err = null
+  for (const modules of debugModules) {
+    try {
+      commonModules.concat(modules)
+        .map(m => require.resolve(m))
+        .forEach(k => delete require.cache[k])
+      err = null
+      break
+    } catch (e) {
+      err = e
+    }
   }
-  deps.forEach((k) => delete require.cache[k])
+  if (err) {
+    throw err
+  }
   process.env.DEBUG = ''
   done()
 })

